hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -55,7 +55,7 @@
   public static final Logger LOG = LoggerFactory.getLogger(AbfsClient.class);
   private final URL baseUrl;
   private final SharedKeyCredentials sharedKeyCredentials;
-  private final String xMsVersion = "2018-11-09";
+  private final String xMsVersion = "2019-12-12";
```

Comment:
Is this version present on all the production clusters already ? If not, this change should go in only after the support is available.

Reply:
nopes actually. how can we go about using the new version with the new API only? is there a way to do this in the driver?

Reply:
Server behaviour is controlled with this API version. So until server is ready, we can not bump up the api version support in driver. 
This is not recommended to be a config control either, to prevent clients from changing api versions to any older one.